content/articles/encapsulation-in-c++/index.md

```diff
@@ -0,0 +1,167 @@
+
+### Encapsulation vs. Inheritance 
+Encapsulation links code to the data it distorts, while inheritance allows a class to inherit the attributes and functions of another class. Encapsulation is exemplified by an ATM that binds together several denominations of banknotes and all of the activities necessary to withdraw money. Inheritance may be seen in the classification of vehicles like cars, bikes, buses, trucks, etc.
+Inheritance means that a subclass gets its primary class's characteristics and functions. On the other hand, encapsulation states that one class should not access another class's private data.
+
+### How Encapsulation is Implemented
+Encapsulation is implemented in C++ as a class that encapsulates data and the methods that operate on it. Data is typically designated as private so that it cannot be accessed outside of the class. The public functions are defined and may be retrieved by using the class's objects.
+However, we cannot openly access private members' data, which is referred to as "data hiding." Data is protected and can be viewed by functions of the class where it is defined when this is performed.
+Below is how it is performed.
+```#include <iostream>
+using namespace std;
+
+class Player {
+  private:
+    // Private attribute
+    int Goals;
+
+  public:
+    // Setter
+    void setGoals(int g) {
+      Goals = g;
+    }
+    // Getter
+    int getGoals() {
+      return Goals;
+    }
+};
+
+int main() {
+  Player myObj;
+  myObj.setGoals(10);
+  cout << myObj.getGoals();
+  return 0;
+}
+```
+The `goals` attribute is private, which has restricted access.
+
+The `public` `setGoals()` method takes a parameter `(g)` and assigns it to the `goals` attribute `(goals = g)`.
+
+The `public getGoals()` method returns the value of the private goals attribute.
+
+Inside `main()`, we create an object of the `Player` class. Now we can use the `setGoals()` method to set the value of the private attribute to **10**. Then we call the `getGoals()` method on the object to return the value.
+
+
+
+### Why Encapsulation
+
+Encapsulation is essential to hold the information about an item hidden from the customers.
+Details of an item are saved in its records individuals. 
+This is the motive we make all of the member variables of a category personal and maximum of the member capabilities public. 
+Member variables are made personal so that those can't be at once accessed from out of doors the magnificence (to cover the info of any item of that magnificence like how the records approximately the item is implemented). So maximum member capabilities are made public to permit the customers to enter the records of individuals via the one's capabilities.
+For example, when it is turned on in an electric kettle, and the water reaches the boiling point, it turns off automatically. We don't know the mechanism behind the kettle turning off, which happens inside it, which is encapsulation.
```

Comment:
when you turn on an electric kettle instead of "when it is turned on in an electric kettle".